src/test/java/io/debezium/connector/yugabytedb/TestHelper.java

```diff
@@ -566,6 +581,23 @@ protected static void waitFor(Duration duration) {
             });
     }
 
+
+    /**
+     * Fail the test if the time exceeds more than 5 minutes to wait for the values.
+     * @param startTime the start time of the verification in milliseconds
+     * @param secondsToWait total duration to wait before failing the test
+     */
+    protected static void failTestIfTimeExceeds(long startTime, long secondsToWait) {
```

Comment:
This function can be replaced with [awaitility + lambdas](). 


For example:

    await().atLeast(1, SECONDS).and().atMost(2, SECONDS).until(value(), equalTo(1));
    
This means that the loops in which it is used will also change. For example:

    await().atmost(10, seconds).until( () -> {
       int consumed = super.consumeAvailableRecords(record -> {
           LOGGER.debug("The record being consumed is " + record);
           records.add(record);
        });
        if (consumed > 0) {
            totalConsumedRecords += consumed;
            LOGGER.debug("Consumed " + totalConsumedRecords + " records");
        }
        return totalConsumedRecords == recordsCount;
     });


If lambda doesn't wok, you can move it into a fn. as well.
Also a polling interval is a good idea.